From: Carlos Garnacho <carlosg@gnome.org>
Date: Mon, 21 Aug 2023 16:53:01 +0200
Subject: clutter: Ensure pick on CLUTTER_PROXIMITY_IN

Same reasoning holds than with touchpad gesture events in the
previous commit, this will be the first event seen from a specific
device and will require early handling, or crashes may follow.

(cherry picked from commit 7a4c98438a928e3c17602c043d2e17fab4d8515b)

Origin: upstream, 44.6, commit:0173cfffcd1fda576877de64a046cad6094fb709
---
 clutter/clutter/clutter-main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/clutter/clutter/clutter-main.c b/clutter/clutter/clutter-main.c
index b2b42f6..9814605 100644
--- a/clutter/clutter/clutter-main.c
+++ b/clutter/clutter/clutter-main.c
@@ -796,6 +796,7 @@ clutter_do_event (ClutterEvent *event)
     case CLUTTER_TOUCHPAD_PINCH:
     case CLUTTER_TOUCHPAD_SWIPE:
     case CLUTTER_TOUCHPAD_HOLD:
+    case CLUTTER_PROXIMITY_IN:
       update_device_for_event (event->any.stage, event, TRUE);
       break;
     default:
